packages/blocks/src/page-block/edgeless/tool-controllers/connector-tool.ts

```diff
@@ -1,39 +1,20 @@
 import type { PointerEventState } from '@blocksuite/block-std';
-import { Rectangle } from '@blocksuite/connector';
 import { assertExists } from '@blocksuite/global/utils';
-import { deserializeXYWH, StrokeStyle } from '@blocksuite/phasor';
+import type { Connection, ConnectorElement, IVec } from '@blocksuite/phasor';
+import { StrokeStyle } from '@blocksuite/phasor';
 
 import type { ConnectorTool } from '../../../__internal__/index.js';
 import { noop } from '../../../__internal__/index.js';
-import {
-  generateConnectorPath,
-  getAttachedPoint,
-} from '../components/connector/utils.js';
-import { getXYWH, pickBy } from '../utils/query.js';
-import type { Selectable, SelectionArea } from '../utils/selection-manager.js';
 import { EdgelessToolController } from './index.js';
 
 export class ConnectorToolController extends EdgelessToolController<ConnectorTool> {
   readonly tool = <ConnectorTool>{
     type: 'connector',
   };
 
-  private _draggingElementId: string | null = null;
-
-  protected override _draggingArea: SelectionArea | null = null;
-  private _draggingStartElement: Selectable | null = null;
-  private _draggingStartRect: Rectangle | null = null;
-  // must assign value when dragging start
-  private _draggingStartPoint!: { x: number; y: number };
-
-  private _pickBy(
-    x: number,
-    y: number,
-    filter: (element: Selectable) => boolean
-  ) {
-    const { surface } = this._edgeless;
-    return pickBy(surface, this._page, x, y, filter);
-  }
+  private _connector: ConnectorElement | undefined = undefined;
```

Comment:
We don't recommend using explicit `undefined` literals, this simply looks weird. Or else use `_connector?` as an alternative.

Reply:
ok, I'll notice that in the following